crates/apps/src/lib/node/ledger/storage/rocksdb.rs

```diff
@@ -554,26 +544,9 @@ impl RocksDB {
         // Restore the state of replay protection to the last block
         let reprot_cf = self.get_column_family(REPLAY_PROTECTION_CF)?;
         tracing::info!("Restoring replay protection state");
-        // Remove the "last" tx hashes
-        for (ref hash_str, _, _) in self.iter_replay_protection() {
-            let hash = namada::core::hash::Hash::from_str(hash_str)
-                .expect("Failed hash conversion");
-            let key = replay_protection::last_key(&hash);
-            batch.delete_cf(reprot_cf, key.to_string());
-        }
-
-        for (ref hash_str, _, _) in self.iter_replay_protection_buffer() {
-            let hash = namada::core::hash::Hash::from_str(hash_str)
-                .expect("Failed hash conversion");
-            let last_key = replay_protection::last_key(&hash);
-            // Restore "buffer" bucket to "last"
-            batch.put_cf(reprot_cf, last_key.to_string(), vec![]);
-
-            // Remove anything in the buffer from the "all" prefix. Note that
-            // some hashes might be missing from "all" if they have been
-            // deleted, this is fine, in this case just continue
-            let all_key = replay_protection::all_key(&hash);
-            batch.delete_cf(reprot_cf, all_key.to_string());
+        // Remove the "current" tx hashes
```

Comment:
Don't we also need to remove the `replay_protection::key(_)` entries committed by the previous block as well?

Reply:
I don't think so. Let's say we are at block 10 and we have tx1, tx2, tx3 committed in the previous blocks and tx4, tx5 committed in 10. Say that we want to rollback to 9, then we need to remove the transactions committed in 10 (tx4, tx5) which live under the `current` subkey.

We could think about bringing the tx committed in block 9 back under the new `current` subkey but I believe this is not needed because these keys are known anyway and the validation doesn't change if the hash lives under the `current` subkey or not. Also, replay protection is not merklized so this is not needed